File: mo-auth/src/main/java/ink/ckx/mo/auth/authentication/mobile/MobileGrantAuthenticationToken.java
package ink.ckx.mo.auth.authentication.mobile;

import ink.ckx.mo.auth.authentication.base.BaseAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.oauth2.core.AuthorizationGrantType;

import java.util.Map;
import java.util.Set;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/06
 */
public class MobileGrantAuthenticationToken extends BaseAuthenticationToken {

    public MobileGrantAuthenticationToken(AuthorizationGrantType authorizationGrantType,
                                          Authentication clientPrincipal, Set<String> scopes, Map<String, Object> additionalParameters) {
        super(authorizationGrantType, clientPrincipal, scopes, additionalParameters);
    }
}

File: mo-auth/src/main/kotlin/ink/ckx/mo/auth/authentication/mobile/MobileGrantAuthenticationToken.kt
package ink.ckx.mo.auth.authentication.mobile

import ink.ckx.mo.auth.authentication.base.BaseAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.core.AuthorizationGrantType

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/06
 */
class MobileGrantAuthenticationToken(
    authorizationGrantType: AuthorizationGrantType,
    clientPrincipal: Authentication,
    scopes: Set<String>,
    additionalParameters: Map<String, Any>
) : BaseAuthenticationToken(authorizationGrantType, clientPrincipal, scopes, additionalParameters)